src/background/messaging/rpc-methods/get-addresses.ts

```diff
@@ -14,11 +16,20 @@ import {
 } from '../messaging-utils';
 import { trackRpcRequestSuccess } from '../rpc-message-handler';
 
-export function makeRpcAddressesMessageListener<T extends RpcMethodNames>(
-  eventName: 'getAddresses' | 'stx_getAddresses'
-) {
-  return async (message: LeatherRpcMethodMap[T]['request'], port: chrome.runtime.Port) => {
-    const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [['requestId', message.id]]);
+export function makeRpcAddressesMessageListener(eventName: 'getAddresses' | 'stx_getAddresses') {
+  return async (
+    message: RpcRequest<typeof getAddresses> | RpcRequest<typeof stxGetAddresses>,
+    port: chrome.runtime.Port
+  ) => {
+    const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [
+      ['requestId', message.id],
+      ['rpcRequest', encodeBase64Json(message)],
+    ]);
+
+    if ('params' in message && message.params && message.params.network) {
```

Comment:
message.params is not enough here?
`if ('params' in message && message.params`

Reply:
Since types updated, no